### PR TITLE
Write Documentation For Predefined Roles [APP-190]

### PR DESCRIPTION
**Pull request by reedcowan**
- Added a Roles section to the Reference guid
- Defined all pre-defined roles in documentation

### CODE REVIEW

File: docs/reference/roles.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+---
+id: roles
+title: Roles
+hide_title: true
+description: Information about pre-defined roles
+keywords:
+ - roles
+ - pre-definted
```
Comment (katthartic) [inline docs/reference/roles.md:8]: Typo

File: docs/reference/roles.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
+---
+id: roles
+title: Roles
+hide_title: true
+description: Information about pre-defined roles
```
Comment (katthartic) [inline docs/reference/roles.md:5]: In the application. Are they displayed as `pre-defined`? Or `predefined`? I think the later... it which case, make this consistent with that.

File: docs/reference/roles.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+## Pre-defined Roles
+
+Roles are our new way of assigning and assessing permissions. While organizations on our Business and Enterprise plans have
+the ability to create their own custom roles and control each permission, organizations on our Team plan only have access to our
+pre-defined roles. Below are in-depth definitions of each pre-defined role and the permissions associated with them.
+
+### Admin
+
+The **Admin** role is just that - **all access to everything**. That includes **Create**, **Read**, **Update**, and **Delete** permissions for:
```
Comment (katthartic) [inline docs/reference/roles.md:21]: And access to update `users`, right? To view, invite, update, delete.

File: docs/reference/roles.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+- **API**
+
+And access to:
+- **User Page** 
+- **Billing Page**
+- **Usage Page**
+
+### Creator
+
+The **Creator** role is a bit more limiting. This **does not include** access to our **User**, **Billing**, or **Usage** page,
```
Comment (katthartic) [inline docs/reference/roles.md:41]: Maybe limited vs limiting?

File: docs/reference/roles.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+Trigger Actions for:
+- **On Demand**
+- **Schedule**
+- **Webhook**
+- **API**
+
+And access to:
+- **User Page** 
+- **Billing Page**
+- **Usage Page**
```
Comment (katthartic) [inline docs/reference/roles.md:37]: And Roles page (if above team)?

File: docs/reference/roles.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+and only provides **Read** permissions for:
+- **Projects**
+- **Blueprints**
+- **Credentials**
+- **Integrations**
+- **Fleets**
+
+
+And Trigger Actions for:
+- **Webhook**
```
Comment (katthartic) [inline docs/reference/roles.md:66]: I thought Viewers couldn't run anything.

File: docs/reference/roles.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+
+### Billing
+
+The **Billing** Role has most of the same limiting permission as our **Viewer** role, however, it **does provide** access to 
+both our **Billing** and **Usage** page. However, the rest of permissions remain the same as the **Viewer** role.
```
Comment (katthartic) [inline docs/reference/roles.md:72]: Suggest changing to... The Billing role has the same permissions as our Viewer role plus access to Usage and Billing information.

File: docs/reference/roles.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+- **Fleets**
+
+And Trigger Actions for:
+- **On Demand**
+- **Webhook**
+- **API**
+
+### Viewer
+
+The **Viewer** role is the most limiting of our roles. This **does not include** access to our **User**, **Billing**, or **Usage**,
```
Comment (katthartic) [inline docs/reference/roles.md:56]: Suggest change to... "is the most limited of the roles"

File: docs/reference/roles.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+
+The **Creator** role is a bit more limiting. This **does not include** access to our **User**, **Billing**, or **Usage** page,
+but does provide Create, Read, and Update permissions for:
+- **Projects**
+- **Blueprints**
+- **Credentials**
+- **Integrations**
+- **Fleets**
+
+And Trigger Actions for:
```
Comment (katthartic) [inline docs/reference/roles.md:49]: Created should have schedule as well, no?

**Review by katthartic — DISMISSED** [review]

**Review by blakeburch — APPROVED** [review]

**Pull request merged**